src/lib/support/FluentTreeObject.h

```diff
@@ -0,0 +1,137 @@
+/*
+ *    Copyright (c) 2024 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <lib/support/Span.h>
+
+#include <optional>
+
+namespace chip {
+
+/// represents a wrapper around a type `T` that contains internal
+/// `Span<...>` values of other sub-types. It allows searching within the container sub-spans
+/// to create new containers.
```

Comment:
My original prototype had it as `class SearchableContainer` but I did not like that either. It does feel like:
   - searches inside trees made out of linear lists (spands)
   - I used `fluent` because it allows `foo.find().find().next()`  (seemed like a common term for method chaining)
   
Will try to figure out better names, however I don't have great ones currently.

Reply:
My current proposal would be something like:
  - `SpanSearchValue` or
  - `SpanValue`

With the argument being:
   - `Value` is the most important thing that we get to at the end
   - This searches within `Span` values as we call find/next/first
 
 Let me know if you have other ideas. Asked an LLM for names and got things like `KeyedSpanIterator`, `HierarchicalSpanNavigator`, `StructuredSpanWalker`, `NestedValueAccessor`, `SpanTreeNavigator`, `FluentSpanSearcher`, `HierarchicalSpanExplorer`, `NestedSpanAccessor`, `KeyedSpanPathfinder`, `SpanChain`, `SpanChainAccessor`, `SpanTraversalValue`, `SpanSearchResult`, `SpanFind` as other ideas/starter names.

Reply:
Renamed to `SpanSearchValue`